lib/hanami/application/routes.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require_relative 'router'
+
+module Hanami
+  class Application
+    # Application routes
+    #
+    # Users are expected to inherit from this class to define their application
+    # routes.
+    #
+    # @example
+    #   # config/routes.rb
+    #   # frozen_string_literal: true
+    #
+    #   require "hanami/application/routes"
+    #
+    #   module MyApp
+    #     class Routes < Hanami::Application::Routes
+    #       slice :main, at: "/" do
+    #         root to: "home.show"
+    #       end
+    #     end
+    #   end
+    #
+    #   See {Hanami::Application::Router} for the syntax allowed at the class
+    #   level.
+    #
+    # @see Hanami::Application::Router
+    # @since 2.0.0
+    class Routes
+      def self.router
+        @router ||= Router.new(
+          resolver: Hanami.application.config.router.resolver.new(
+            slices: Hanami.application.slices,
+            inflector: Hanami.application.inflector
+          ),
+          **Hanami.application.configuration.router.options,
+        ) do
+          use Hanami.application[:rack_monitor]
```

Comment:
@waiting-for-dev Please remove those middleware from here and add them to the Rack middleware stack of the application.

In depth explanation. We can have three nested layers of Rack middleware stacks. Here's them listed from the outermost to the innermost:

* `config.ru`
* Application
* Router

The less stack we setup the better, for the app performance. When adding default middleware to the stack, we should target Application.

Reply:
I think it's food for another PR :slightly_smiling_face: